utils/metadata_generator.py

```diff
@@ -17,19 +17,32 @@ def extract_macros_params(line, macros_name):
         return extracted
     return None
 
-def parse_declare_saved_begin(file, class_and_base, filename):
+
+def parse_file(full_path: str) -> Union[dict, None]:
+    with open(full_path, 'r') as file:
+        for line in file:
+            result = extract_macros_params(line, "DECLARE_SAVEABLE")
+            if result:
+                data = parse_declare_saved_begin(file, result, full_path)
+                return data
+        return None
```

Comment:
Now it fails to compile because metadata is not generated for each class.
There might be more than one `DECLARE_SAVEABLE` per file (multiple game classes in one header), and seems like now only first one will be processed.
E.g. https://github.com/fluorescent-net/griefly/blob/master/sources/core/objects/movable/Pipes.h file contains multiple `DECLARE_SAVEABLE`.